### PR TITLE
vm change to reproducible command

### PR DESCRIPTION
**Pull request by olafklingt**
add paragraph on reproducible nixos-generate-config command

### CODE REVIEW

File: source/tutorials/nixos/nixos-configuration-on-vm.md [hunk 1]
```diff
@@ -20,14 +20,22 @@ Virtual machines are a practical tool for debugging NixOS configurations.
...
 
-In this tutorial you will use the default configuration that is shipped with NixOS.[^nixosconf]
-[^nixosconf]: This [configuration template](https://github.com/NixOS/nixpkgs/blob/4e0525a8cdb370d31c1e1ba2641ad2a91fded57d/nixos/modules/installer/tools/tools.nix#L122-L226) is used.
+In this tutorial you will use a default configuration that is shipped with NixOS.
 
 :::{admonition} NixOS
 
 On NixOS, use the `nixos-generate-config` command to create a configuration file that contains some useful defaults and configuration suggestions.
+
+The result of this command depends on your curent Nixos configuration.
```
Comment (fricklerhandwerk) [inline source/tutorials/nixos/nixos-configuration-on-vm.md:31]:
```suggestion
Beware that the result of this command depends on your current NixOS configuration.
```

:sob: But why does it have to be this way!

File: source/tutorials/nixos/nixos-configuration-on-vm.md [hunk 2]
```diff
@@ -20,14 +20,22 @@ Virtual machines are a practical tool for debugging NixOS configurations.
...
+We can configure the generation of the 'nixos-generate-config' in a repoducable way in a nix-shell. 
+We provide a configuration that is replicating nixos gnome graphical iso image.
```
Comment (fricklerhandwerk) [inline source/tutorials/nixos/nixos-configuration-on-vm.md]:
```suggestion
The output of  'nixos-generate-config' can be made reproducible in a `nix-shell` environment. 
Here we provide a configuration that corresponds to the NixOS GNOME graphical ISO image:
```

File: source/tutorials/nixos/nixos-configuration-on-vm.md [hunk 3]
```diff
@@ -20,14 +20,22 @@ Virtual machines are a practical tool for debugging NixOS configurations.
...
+
+```shell-session
+nix-shell -p "with (import <nixpkgs> {}); (nixos { services.xserver.enable = true; services.xserver.desktopManager.gnome.enable = true;}).config.system.build.nixos-generate-config" -I nixpkgs=channel:nixos-21.05
+```
+
```
Comment (fricklerhandwerk) [inline source/tutorials/nixos/nixos-configuration-on-vm.md:38]: This is a true abomination, which shows that Nix is awesome despite NixOS being in a very sad place. I love it. @olafklingt what a great piece of tinkering!

Suggestion to remove the veil of magic:

````suggestion

```bash
nix-shell -I nixpkgs=channel:nixos-21.05 -p "$(cat <<EOF
let
  pkgs = import <nixpkgs> { config = {}; overlays = []; };
  nixos = pkgs.nixos {
    services.xserver.enable = true;
    services.xserver.desktopManager.gnome.enable = true;
  };
in
nixos.config.system.build.nixos-generate-config
EOF
)" 
```

Or as a one-liner:

```shell-session
nix-shell -I nixpkgs=channel:nixos-21.05 -p "let pkgs = import <nixpkgs> { config = {}; overlays = []; }; nixos = pkgs.nixos { services.xserver.enable = true; services.xserver.desktopManager.gnome.enable = true; }; in nixos.config.system.build.nixos-generate-config"
```

````

**Review by fricklerhandwerk — APPROVED** [review]
This is just insane. What a cool permanent solution to make the tutorial reproducible!

File: source/tutorials/nixos/nixos-configuration-on-vm.md [hunk 4]
```diff
@@ -20,14 +20,37 @@ Virtual machines are a practical tool for debugging NixOS configurations.
...
-[^nixosconf]: This [configuration template](https://github.com/NixOS/nixpkgs/blob/4e0525a8cdb370d31c1e1ba2641ad2a91fded57d/nixos/modules/installer/tools/tools.nix#L122-L226) is used.
+In this tutorial you will use a default configuration that is shipped with NixOS.
 
 :::{admonition} NixOS
 
 On NixOS, use the `nixos-generate-config` command to create a configuration file that contains some useful defaults and configuration suggestions.
+
+Beware that the result of this command depends on your current NixOS configuration.
+We can configure the generation of the 'nixos-generate-config' in a repoducable way in a nix-shell.
+We provide a configuration that is replicating nixos gnome graphical iso image.
```
Comment (fricklerhandwerk) [inline source/tutorials/nixos/nixos-configuration-on-vm.md:33]: Hm, apparently that suggestion got lost in a merge conflict. Sorry for the second round:

```suggestion
The output of 'nixos-generate-config' can be made reproducible in a `nix-shell` environment. 
Here we provide a configuration that corresponds to the NixOS GNOME graphical ISO image:
```

**Comment by olafklingt** [comment]
I think the reproducable command could be improved if we find a way use the nixos configuration from the installer as it is 
defined here
https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome/installer.nix
or here:
https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix
but i don't know how to do that.

**Comment by fricklerhandwerk** [comment]
@Roberth spontaneous ideas for follow ups?

Other than that, awesome stuff, merging.

**Pull request merged**